extensions/amp-wpm-player/0.1/amp-wpm-player.js

```diff
@@ -0,0 +1,557 @@
+/**
+* Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS-IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import {
+  PlayingStates,
+  VideoAttributes,
+  VideoEvents,
+} from '../../../src/video-interface';
+import {Services} from '../../../src/services';
+
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+
+import {
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+} from '../../../src/dom';
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+export class AmpWpmPlayer extends AMP.BaseElement {
+  /**
+   * @description Method that parses attributes,
+   * and ensures that all of the required parameters are present
+   * @function
+   * @private
+   *
+   * @return {Object}
```

Comment:
If you define a type for this attribute set you can get type checking and it would be easier to find simple usage errors without unit tests.

```js
/**
  * @typedef {{
  *  ampcontrols: boolean,
  *  autoplay: boolean,
  *  forceUrl4stat: string,
  *  target: string,
  *  clip: (!JsonObject|undefined),
  *  adv: (boolean|undefined),
  *  url: (string|undefined),
  *  title: (string|undefined),
  *  screenshot: (string|undefined),
  *  forcerelated: (boolean|undefined)
  *  hiderelated: (boolean|undefined)
  *  hideendscreen: (boolean|undefined)
  *  mediaEmbed: (string|undefined),
  *  extendedrelated: (boolean|undefined)
  *  skin: (!JsonObject|undefined),
  *  showlogo: (boolean|undefined)
  *  watermark: (boolean|undefined)
  *  qoeEventsConfig: (!JsonObject|undefined),
  *  advVastDuration: (number|undefined),
  *  vastTag: (string|undefined),
  *  embedTrackings: (!JsonObject|undefined),
  *  id: (string|undefined),
  *  ampnoaudio: (boolean|undefined)
  *  dock: (boolean|undefined),
  *  rotateToFullscreen: (boolean|undefined),
  * }}
  */
let AttributeOptionsDef;

// ...

/**
 * ...
 * @return {!AttributeOptionsDef}
 */
```